pipeline/src/infer/generate_inference_results.jl

```diff
@@ -2,18 +2,20 @@
 Generate inference results based on the given configuration of inference model options.
 
 # Arguments
+- `truthdata`: The truth data used for generating inference results.
 - `inference_config`: A dictionary containing the inference configuration choices.
+- `pipeline::AbstractEpiAwarePipeline`: The pipeline object which sets pipeline
+    behavior. This is the default method.
 - `tspan`: The time span for the inference.
 - `inference_method`: The method used for inference.
-- `truth_data_config`: A dictionary containing the truth data configuration parameters. This is only
-    used to generate the file name for the inference results.
+
 
 # Returns
 - `inference_results`: The generated inference results.
-- `inferencefile`: The file path where the inference results are stored.
 """
-function generate_inference_results(truthdata, inference_config; tspan, inference_method,
-        truth_data_config, latent_models_names,
+function generate_inference_results(
+        truthdata, inference_config, pipeline::AbstractEpiAwarePipeline;
+        tspan, inference_method, latent_models_names,
```

Comment:
> why is latent_model_names an arguement? Why isnt it part of the latent config?

So the slight issue here is that I'm using `DrWatson.dict_list` to spool out the possible configs. Done in a naive way you'd get a loop over models and names of models, hence how it is here because this way does the logic:

- using a latent model
- look for name assigned to model

However, I'm sure there is a less clunky way of doing this (e.g. spool out a vector of `(;modelname, model)`)

Reply:
I'm not sure I totally follow? Why can't the full name be part of the dict? IMO this doesn't need to delay this PR and can be its own issue.

Reply:
I've fixed it now.

The specific matter was the way `dict_list` unspools a dictionary with vector valuetypes into a vector of dicts.

My fix is that the latentmodels are now passed as vector of `Pair`s in the form `name => latentmodel`. This is easy to work with by passing to `generate_inference_results` because the `second` field of the `Pair` goes to the inference and the `first` field goes to the prefix string bit.

Reply:
Got it. Nice solution